espGarden/main/include/wifi_manager.h

```diff
@@ -87,34 +36,71 @@ esp_err_t wifi_manager_get_ip(char* ip){
     return ESP_OK;
 }
 
-void wifi_task(void *pvParameters) {
+// Task prints 
+void wifi_print_stats(void *pvParameters) {
     wifi_ap_record_t ap_info;
     esp_netif_ip_info_t ip;
     memset(&ip, 0, sizeof(esp_netif_ip_info_t));
-    vTaskDelay(2000 / portTICK_PERIOD_MS);
 
+    // Refresh the wifi setup until its connected
     while (1) {
+        // Check for connection every 5 seconds
         vTaskDelay(5000 / portTICK_PERIOD_MS);
+        bool is_connected = (esp_netif_get_ip_info(sta_netif, &ip) == 0 && esp_wifi_sta_get_ap_info(&ap_info) == ESP_OK);
 
-        if (esp_netif_get_ip_info(sta_netif, &ip) == 0 && esp_wifi_sta_get_ap_info(&ap_info) == ESP_OK) {
+        // Check/print connection every 10 minutes
+        if (is_connected) {
             ESP_LOGI(WIFI_LOG_TAG, "~~~~~~~~~~~");
```

Comment:
WIFI_LOG_TAG is coming from heltec_unofficial how can we access it here? was this tested am i missing something.

Reply:
The log tags are declared `static` which puts them in the global scope, accessible to all functions

Reply:
are u sure. I swear static keyword restricts the scope of the variable to the file in which it is declared.